### ISSUE TITLE
Error on persistent action while helm-gtags-select

### ISSUE THREAD

**Issue by bassu** [syohex/emacs-helm-gtags#142]
All I am doing is pressing `C-j` or in other words the `tab` and persistent action fails.
It works manually with `C-z` and then choosing `f2`; or by pressing `return`.

Here's the backtrace:

```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  string-to-number(nil)
  helm-gtags--extract-file-and-line("print_frm_indicies")
  helm-gtags--persistent-action("print_frm_indicies")
  helm-execute-selection-action-1(nil helm-gtags--persistent-action t)
  helm-execute-persistent-action()
  call-interactively(helm-execute-persistent-action nil nil)
  command-execute(helm-execute-persistent-action)
  read-from-minibuffer("pattern: " nil (keymap (tab . helm-execute-persistent-action) (f13 lambda nil (interactive) (helm-select-nth-action 12)) (f12 lambda nil (interactive) (helm-select-nth-action 11)) (f11 lambda nil (interactive) (helm-select-nth-action 10)) (f10 lambda nil (interactive) (helm-select-nth-action 9)) (f9 lambda nil (interactive) (helm-select-nth-action 8)) (f8 lambda nil (interactive) (helm-select-nth-action 7)) (f7 lambda nil (interactive) (helm-select-nth-action 6)) (f6 lambda nil (interactive) (helm-select-nth-action 5)) (f5 lambda nil (interactive) (helm-select-nth-action 4)) (f4 lambda nil (interactive) (helm-select-nth-action 3)) (f3 lambda nil (interactive) (helm-select-nth-action 2)) (f2 lambda nil (interactive) (helm-select-nth-action 1)) (menu-bar keymap (help-menu keymap (describe keymap (describe-mode . helm-help)))) (help keymap (109 . helm-help)) (f1 lambda nil (interactive) (helm-select-nth-action 0)) (8 keymap (109 . helm-help) (104 . undefined) (8 . undefined) (4 . helm-enable-or-switch-to-debug)) (20 . helm-toggle-resplit-and-swap-windows) (C-tab . undefined) (67108897 . helm-toggle-suspend-update) (3 keymap (57 lambda nil (interactive) (helm-execute-selection-action-at-nth 9)) (56 lambda nil (interactive) (helm-execute-selection-action-at-nth 8)) (55 lambda nil (interactive) (helm-execute-selection-action-at-nth 7)) (54 lambda nil (interactive) (helm-execute-selection-action-at-nth 6)) (53 lambda nil (interactive) (helm-execute-selection-action-at-nth 5)) (52 lambda nil (interactive) (helm-execute-selection-action-at-nth 4)) (51 lambda nil (interactive) (helm-execute-selection-action-at-nth 3)) (50 lambda nil (interactive) (helm-execute-selection-action-at-nth 2)) (49 lambda nil (interactive) (helm-execute-selection-action-at-nth 1)) (63 . helm-help) (62 . helm-toggle-truncate-line) (21 . helm-refresh) (6 . helm-follow-mode) (9 . helm-copy-to-buffer) (11 . helm-kill-selection-and-quit) (25 . helm-yank-selection) (4 . helm-delete-current-selection) (45 . helm-swap-windows)) (67108987 . helm-enlarge-window) (67108989 . helm-narrow-window) (19 . undefined) (18 . undefined) (23 . helm-yank-text-at-point) (24 keymap (57 lambda nil (interactive) (helm-execute-selection-action-at-nth -9)) (56 lambda nil (interactive) (helm-execute-selection-action-at-nth -8)) (55 lambda nil (interactive) (helm-execute-selection-action-at-nth -7)) (54 lambda nil (interactive) (helm-execute-selection-action-at-nth -6)) (53 lambda nil (interactive) (helm-execute-selection-action-at-nth -5)) (52 lambda nil (interactive) (helm-execute-selection-action-at-nth -4)) (51 lambda nil (interactive) (helm-execute-selection-action-at-nth -3)) (50 lambda nil (interactive) (helm-execute-selection-action-at-nth -2)) (49 lambda nil (interactive) (helm-execute-selection-action-at-nth -1)) (2 . helm-resume-list-buffers-after-quit) (98 . helm-resume-previous-session-after-quit) (6 . helm-quit-and-find-file)) (11 . helm-delete-minibuffer-contents) (67108896 . helm-toggle-visible-mark) (0 . helm-toggle-visible-mark) (C-M-up . helm-scroll-other-window-down) (C-M-down . helm-scroll-other-window) (M-prior . helm-scroll-other-window-down) (M-next . helm-scroll-other-window) (12 . helm-recenter-top-bottom-other-window) (15 . helm-next-source) (10 . helm-execute-persistent-action) (26 . helm-select-action) (9 . helm-execute-persistent-action) (13 . helm-maybe-exit-minibuffer) (left . helm-previous-source) (right . helm-next-source) (7 . helm-keyboard-quit) (22 . helm-next-page) (27 keymap (110 . next-history-element) (112 . previous-history-element) (115 . undefined) (5 . helm-display-all-sources) (1 . helm-show-all-in-this-source-only) (85 . helm-unmark-all) (97 . helm-mark-all) (109 . helm-toggle-all-marks) (41 . helm-next-visible-mark) (40 . helm-prev-visible-mark) (91) (32 . helm-toggle-visible-mark) (33554454 . helm-scroll-other-window-down) (25 . helm-scroll-other-window-down) (22 . helm-scroll-other-window) (12 . helm-reposition-window-other-window) (62 . helm-end-of-buffer) (60 . helm-beginning-of-buffer) (118 . helm-previous-page)) (next . helm-next-page) (prior . helm-previous-page) (C-up . helm-follow-action-backward) (C-down . helm-follow-action-forward) ...) nil nil #("void" 0 1 (fontified t c-is-sws t face font-lock-type-face) 1 3 (fontified t face font-lock-type-face) 3 4 (fontified t c-type c-decl-id-start face font-lock-type-face)) t)
  helm-read-pattern-maybe(nil nil "util\\.c:81" nil nil nil nil)
  #[0 "\311\211\211\211\305\206\n&&'()*\312\313+\"\314\315\316\317\320\321!\322\"\323\324%DC\216\325\303\301\307\300$\210\326&!\210,\203?\327\330!\210\331\332\333\"\210\331\332\334\"\210\335\336!\210\314\337DC\216\340\302\301\304\303\306\307\310&\210*\210'?\205g\341 \335\342\343\344\345\"P!\210-\207" [(helm-source-gtags-select) nil nil nil "util\\.c:81" "*helm gtags*" nil nil nil nil mapcar #[257 "\211JB\207" [] 3 "\n\n(fn V)"] funcall make-byte-code 0 "\300\301\211:\203@\262\211A\262\242\262\211L\210A\262\202\266\302\303!\207" vconcat vector [nil helm-log "restore variables"] 5 "\n\n(fn)" helm-initialize helm-display-buffer helm--remap-mouse-mode 1 add-hook post-command-hook helm--maybe-update-keymap helm--update-header-line helm-log "show prompt" #[0 "\300 \207" [helm-cleanup] 1 "\n\n(fn)"] helm-read-pattern-maybe helm-execute-selection-action "[End session] " make-string 41 45 helm-buffer helm-quit helm-in-persistent-action helm-current-source helm-source-name helm-restored-variables helm-prevent-escaping-from-minibuffer] 9 "\n\n(fn)"]()
  funcall(#[0 "\311\211\211\211\305\206\n&&'()*\312\313+\"\314\315\316\317\320\321!\322\"\323\324%DC\216\325\303\301\307\300$\210\326&!\210,\203?\327\330!\210\331\332\333\"\210\331\332\334\"\210\335\336!\210\314\337DC\216\340\302\301\304\303\306\307\310&\210*\210'?\205g\341 \335\342\343\344\345\"P!\210-\207" [(helm-source-gtags-select) nil nil nil "util\\.c:81" "*helm gtags*" nil nil nil nil mapcar #[257 "\211JB\207" [] 3 "\n\n(fn V)"] funcall make-byte-code 0 "\300\301\211:\203@\262\211A\262\242\262\211L\210A\262\202\266\302\303!\207" vconcat vector [nil helm-log "restore variables"] 5 "\n\n(fn)" helm-initialize helm-display-buffer helm--remap-mouse-mode 1 add-hook post-command-hook helm--maybe-update-keymap helm--update-header-line helm-log "show prompt" #[0 "\300 \207" [helm-cleanup] 1 "\n\n(fn)"] helm-read-pattern-maybe helm-execute-selection-action "[End session] " make-string 41 45 helm-buffer helm-quit helm-in-persistent-action helm-current-source helm-source-name helm-restored-variables helm-prevent-escaping-from-minibuffer] 9 "\n\n(fn)"])
  helm-internal((helm-source-gtags-select) nil nil nil "util\\.c:81" "*helm gtags*" nil nil nil)
  apply(helm-internal ((helm-source-gtags-select) nil nil nil "util\\.c:81" "*helm gtags*" nil nil nil))
  helm((helm-source-gtags-select) nil nil nil "util\\.c:81" "*helm gtags*" nil nil nil)
  apply(helm ((helm-source-gtags-select) nil nil nil "util\\.c:81" "*helm gtags*" nil nil nil))
  helm(:sources (helm-source-gtags-select) :buffer "*helm gtags*" :preselect "util\\.c:81")
  helm-gtags--common((helm-source-gtags-select) nil)
  helm-gtags-select()
  call-interactively(helm-gtags-select record nil)
  command-execute(helm-gtags-select record)
  helm-M-x(nil "helm-gtags-select")
  call-interactively(helm-M-x nil nil)
  command-execute(helm-M-x)
```

**Comment by syohex** [comment]
Thanks for reporting issue. I set wrong persistent action to `helm-gtags-select`(In fact, `helm-gtags-select` does not support persistent action). I have fixed this issue at #143. Please check latest version.

**Comment by bassu** [comment]
Thanks for the quick fix!  :japanese_castle:
